subsystems/TypeHandlerLibrary/src/main/java/org/terasology/persistence/typeHandling/coreTypes/factories/ObjectFieldMapTypeHandlerFactory.java

```diff
@@ -77,7 +83,18 @@ private <T> Map<Field, Type> getResolvedFields(TypeInfo<T> typeInfo) {
                         continue;
                     }
 
-                    field.setAccessible(true);
+                    if (Arrays.stream(field.getAnnotations())
+                            .anyMatch(annotation -> PRIVATE_OVERRIDE_ANNOTATIONS.contains(annotation.getClass().getCanonicalName()))) {
+                        field.setAccessible(true);
+                    } else {
+                        if (Modifier.isPrivate(field.getModifiers()) && ReflectionUtil.findGetter(field) == null
+                                && ReflectionUtil.findSetter(field) == null) {
+                            logger.atWarn().addArgument(field.getName()).addArgument(rawType.getTypeName()).
+                                    log("Field {}#{} will not be serialised. Terasology no longer supports serialising private fields.");
```

Comment:
Why fluent API usage here?
I would expect `getName()` and `getTypeName()` to be O(1) so pretty much neglectable... also this is not on the happy path so it's not expected to be evaluated often...
In my test run, I got the warning only twice:
```
19:18:19.310 [main] WARN  o.t.p.t.c.f.ObjectFieldMapTypeHandlerFactory - Field exists#org.terasology.engine.entitySystem.entity.internal.PojoEntityRef will not be serialised. Terasology no longer supports serialising private fields.
19:18:19.396 [parallel-1] WARN  o.t.p.t.c.f.ObjectFieldMapTypeHandlerFactory - Field entityRef#org.terasology.engine.persistence.internal.DelayedEntityRef will not be serialised. Terasology no longer supports serialising private fields.
```

Reply:
okay, guess it might've been a PMD warning in that case.
I'd prefer suppressing the warning (suffix with `//NOPMD`) here due to the above-mentioned reasons.

Reply:
i am a little confused  - git blame says @BenjaminAmos changed that line. commit 873fe5b changed ComponentTypeHandlerFactory, different file, but the change looks the same? in case i really did this, feel free to undo. 2 warnings, one line too long, and one for the logger are there in the line before.